### ISSUE TITLE
Error (1621) on Bose Soundtouch 30 apps

### ISSUE THREAD

**Issue by mergality** [gerbera/gerbera#1080]
Hi,

I used to run gerbera 1.3.5 on Mint 19.3 (Ubuntu bionic) fine, but after upgrading to Mint 20 (Ubuntu focal) came version 1.6.0 (btw. both from ppa:stephenczetty/gerbera-update) I have issues when browsing most database trees from Bose Sountouch app. I get Error (1621) - like [here](https://community.bose.com/t5/SoundTouch-Archive/UPnP-with-Soundtouch-30/td-p/220160).

I don't have any issues while browsing in e.g. VLC.

Someone mentioned [here](https://forum.minimserver.com/showthread.php?tid=5487) they got this error with MinimServer possibly due to too long URLs but that shouldn't be an issue with gerbera as it uses database indexes for URLs.

Strangely I don't get this issue (at least so far) while browsing via Genre tree and it's hit or miss while browsing via Year (I can always get up to a point of individual years listed in a decade, but sometimes not further).

Now I'm running 1.3.2 from  deb http://www.deb-multimedia.org buster main. I had to remove and rebuild the database to get rid of the error.
I got the same issue when running database created 1.6.4 from deb http://www.deb-multimedia.org bullseye main.

My import script (only slightly modified the default one):
`
function addAudio(obj) {
 
    var desc = '';
    var artist_full;
    var album_full;

    var decade = null;
    
    // first gather data
    var title = obj.meta[M_TITLE];
    if (!title) title = obj.title;
/*  
    var artist = obj.meta[M_ARTIST];
    if (!artist) {
        artist = 'Unknown';
        artist_full = null;
    } else {
        artist_full = artist;
        desc = artist;
    }
*/

//Wlasne
    var location = obj.location.split("/");
    if (location.length == 5) {
        var artist = location[2];
        var album = location[3];
        //var al = album.split(") ");
        //if (al.length == 2) album = al[1];
        album_full = album;
    } else {
        var artist = location[2];
        var album = location[2];
        //var al = album.split(") ");
        //if (al.length == 2) album = al[1];
        album_full = null;    
    }

    artist_full = artist;
    desc = artist;
    desc = desc + ', ' + album;
    
/*
    var album = obj.meta[M_ALBUM];
    if (!album) {
        album = 'Unknown';
        album_full = null;
    } else {
        desc = desc + ', ' + album;
        album_full = album;
    }

 */

    if (desc) {
        desc = desc + ', ';
    }
    
    desc = desc + title;
    
    var date = obj.meta[M_DATE];
    if (!date) {
        date = '-Unknown-';
        decade = '-Unknown-';
    }
    else {
        date = getYear(date);
        obj.meta[M_UPNP_DATE] = date;
        desc = desc + ', ' + date;
        decade = date.substring(0,3) + '0 - ' + String(10 * (parseInt(date.substring(0,3))) + 9) ;
    }
    
    var genre = obj.meta[M_GENRE];
    if (!genre) {
        genre = 'Unknown';
    } else {
        desc = desc + ', ' + genre;
    }
    
    var description = obj.meta[M_DESCRIPTION];
    if (!description) {
        obj.meta[M_DESCRIPTION] = desc;
    }

// uncomment this if you want to have track numbers in front of the title
// in album view
    
 /*  
    var track = obj.meta[M_TRACKNUMBER];
    if (!track)
        track = '';
    else
    {
        if (track.length == 1)
        {
            track = '0' + track;
        }
        track = track + ' ';
    }
*/
    // comment the following line out if you uncomment the stuff above  :)
    var track = '';

    
    // Album
    
    // Extra code for correct display of albums with various artists (usually collections)
    if (!description) {
        //album_artist = album + ' - ' + artist;
        album_artist = artist + ' - ' + album;
        tracktitle = track + title;
    } else {
        if (description.toUpperCase() === 'VARIOUS') {
            album_artist = album + ' - Various';
            tracktitle = track + title + ' - ' + artist;
        } else {
            album_artist = album + ' - ' + artist;
            tracktitle = track + title;
        }
    }

    chain = ['-Album-', abcbox(album, 6, '-'), album.charAt(0).toUpperCase(), album_artist];
    obj.title = tracktitle;
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ALBUM);

    chain = ['-Album-', abcbox(album, 6, '-'), '-all-', album_artist];
    obj.title = tracktitle;
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ALBUM);

    chain = ['-Album-', '--all--', album_artist];
    obj.title = tracktitle;
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ALBUM);

    // Artist

    chain = ['-Artist-',  '--all--', artist];
    obj.title = album +' - '+ title + ' (' + date + ')';
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ARTIST);

    chain = ['-Artist-', abcbox(artist, 9, '-'), '-all-', artist];
    obj.title = album +' - '+ title + ' (' + date + ')';
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ARTIST);

    chain = ['-Artist-', abcbox(artist, 9, '-'), artist.charAt(0).toUpperCase(), artist, '-all-'];
    obj.title = album +' - '+ title + ' (' + date + ')';
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ARTIST);

    obj.title = tracktitle;
    chain = ['-Artist-', abcbox(artist, 9, '-'), artist.charAt(0).toUpperCase(), artist, album + ' (' + date + ')'];
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ALBUM);

    // Genre

    chain = ['-Genre-', genre, '--all--'];
    obj.title = title + ' - ' + artist_full;
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_GENRE);

    chain = ['-Genre-', genre, abcbox(artist, 6, '-'), artist.charAt(0).toUpperCase(), album_artist];
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ALBUM);

    // Tracks
/*
    chain = ['-Track-', abcbox(title, 6, '-'), title.charAt(0).toUpperCase()];
    obj.title = title + ' - ' + artist + ' (' + album + ', ' + date + ')';
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ARTIST);

    chain = ['-Track-', '--all--'];
    obj.title = title + ' - ' + artist_full;
    addCdsObject(obj, createContainerChain(chain));

*/
    // Sort years into decades

    chain = ['-Year-', decade, '-all-'];
    addCdsObject(obj, createContainerChain(chain));

    chain = ['-Year-', decade, date, '-all-'];
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ARTIST);

    chain = ['-Year-', decade, date, artist, album];
    obj.title = tracktitle;
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER_MUSIC_ALBUM);
}
/*
function addVideo(obj) {
    var chain = ['Video', 'All Video'];
    addCdsObject(obj, createContainerChain(chain));

    var dir = getRootPath(object_script_path, obj.location);

    if (dir.length > 0) {
        chain = ['Video', 'Directories'];
        chain = chain.concat(dir);

        addCdsObject(obj, createContainerChain(chain));
    }
}

function addImage(obj) {
    var chain = ['Photos', 'All Photos'];
    addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER);

    var date = obj.meta[M_DATE];
    if (date) {
        var dateParts = date.split('-');
        if (dateParts.length > 1) {
            var year = dateParts[0];
            var month = dateParts[1];

            chain = ['Photos', 'Year', year, month];
            addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER);
        }

        chain = ['Photos', 'Date', date];
        addCdsObject(obj, createContainerChain(chain), UPNP_CLASS_CONTAINER);
    }

    var dir = getRootPath(object_script_path, obj.location);

    if (dir.length > 0) {
        chain = ['Photos', 'Directories'];
        chain = chain.concat(dir);

        addCdsObject(obj, createContainerChain(chain));
    }
}

function addTrailer(obj) {
    var chain;

    chain = ['Online Services', 'Apple Trailers', 'All Trailers'];
    addCdsObject(obj, createContainerChain(chain));

    var genre = obj.meta[M_GENRE];
    if (genre) {
        genres = genre.split(', ');
        for (var i = 0; i < genres.length; i++) {
            chain = ['Online Services', 'Apple Trailers', 'Genres', genres[i]];
            addCdsObject(obj, createContainerChain(chain));
        }
    }

    var reldate = obj.meta[M_DATE];
    if ((reldate) && (reldate.length >= 7)) {
        chain = ['Online Services', 'Apple Trailers', 'Release Date', reldate.slice(0, 7)];
        addCdsObject(obj, createContainerChain(chain));
    }

    var postdate = obj.aux[APPLE_TRAILERS_AUXDATA_POST_DATE];
    if ((postdate) && (postdate.length >= 7)) {
        chain = ['Online Services', 'Apple Trailers', 'Post Date', postdate.slice(0, 7)];
        addCdsObject(obj, createContainerChain(chain));
    }
}
*/
// main script part

if (getPlaylistType(orig.mimetype) === '') {
    var arr = orig.mimetype.split('/');
    var mime = arr[0];
    
    // var obj = copyObject(orig);
    
    var obj = orig; 
    obj.refID = orig.id;
    
    if (mime === 'audio') {
            addAudio(obj);
    }
    
/*    if (mime === 'video') {
        if (obj.onlineservice === ONLINE_SERVICE_APPLE_TRAILERS) {
            addTrailer(obj);
        } else {
            addVideo(obj);
        }
    }
*/    
/*    if (mime === 'image') {
        addImage(obj);
    }
*/
/*   if (orig.mimetype === 'application/ogg') {
        if (orig.theora === 1) {
            addVideo(obj);
        } else {
            addAudio(obj);
        }
    }
*/
}`

To sum it up: with databases created in 1.3.2 and 1.3.5 I don't get the error, but I get it on 1.6.0 and 1.6.4.

Any ideas for a workaround?

**Comment by KarlStraussberger** [comment]
1.6.0 contained a couple of issues which took some further patches. So try to get hold of 1.6.4 if possible.

**Comment by KarlStraussberger** [comment]
There were changes to search in 1.8.2 and UPNP which might also fix your issue. Please reopen issue, if the error still exists
